array.c

```diff
@@ -5767,19 +5767,14 @@ rb_ary_union_multi(int argc, VALUE *argv, VALUE ary)
 
 /*
  *  call-seq:
- *     ary.intersect?(other_ary)   -> true or false
+ *    intersect?(other_array) -> true or false
  *
- *  Returns +true+ if the array and +other_ary+ have at least one element in
- *  common, otherwise returns +false+:
+ *  Returns whether +other_array+ has at least one element that is #eql? to some element of +self+:
```

Comment:
Nit: we don't want to link to `Array#eql?` (which I don't think even exists), but rather for it to refer to the method `eql?` for the elements in the array.

```suggestion
 *  Returns whether +other_array+ has at least one element that is +#eql?+ to some element of +self+:
```